### ISSUE TITLE
Buttons don't meet WCAG AA on lightmode

### ISSUE THREAD

**Issue by Asartea** [TheOdinProject/theodinproject#3923]
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this issue follows the `brief description of request` format, e.g. `Add dark mode to website`

The following checkbox is OPTIONAL:

-   [x] I would like to be assigned this issue to work on it

<hr>

**1. Description of the Feature Request:**
Splitting of from #3918; buttons suffer from the same contrast issue on light mode as links. In that issues, @KevinMulhern said:
> I missed buttons being mentioned in the issue. It could maybe be split out into its own issue. It does have the same underlying issue, but affects different UI components.

> Buttons should thankfully be straight forward though. No color/transparent buttons work best with secondary UI actions

While I agree in principle that no color/transparent buttons are how we should handle secondary actions, that brings up the question of which of the three buttons at the bottom of a lesson is the primary action; while "mark as completed" is probably right to be marked as such, I can also see a argument for "Next lesson", especially as finishing a lesson requires pressing both (radical idea: Make mark as completed also redirect the user to the next lesson)

**2. Acceptance Criteria:**
- [ ] Decide how to handle these buttons
- [ ] Implement this

**Comment by KevinMulhern** [comment]
For me, "mark completed" is the primary thing you want to do when you reach the end of the lesson and the other buttons are navigational / less important. 

I see what you mean about next lesson, it probably is the next thing most will do after completing a lesson. I don't believe it needs to be made more obvious though. It's proximity and position to the right of the complete button makes it intuitive and easy to find.

**Comment by Asartea** [comment]
Something like this maybe? 
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/f6c84f4a-ea78-49cb-bfe3-615944bbee79)
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/3f6e7ca9-cfbd-47f7-9c80-ea629b1dfb91)

**Comment by KevinMulhern** [comment]
That looks like the button has focus - I'd be inclined to do something extra if we were getting any feedback about the next action being difficult to find/understand, but it's never been an issue. I don't think we should overthink this and just go with plain transparent buttons.

**Comment by Asartea** [comment]
Not sure which you meant, but the right button on dark mode / left button on light mode is focused yes (to show how that would look). Don't ask me why its a different button for each

**Comment by KevinMulhern** [comment]
lol sorry, my bad. Yep, those great to me 🔥

**Comment by Asartea** [comment]
Actually, we can just reuse `button--clear` for the lesson buttons, which looks like this:
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/fefd14b5-602f-4a99-97e7-f0eb692f1132)
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/5247b0bc-a7a8-4051-94ce-0dd967c93edf)



By the cunning work of typing `button--secondary` into VSCode search it seems the other uses are:

1. the view full curriculum button on the home page
2. the explore curriculum button on the same page
3. the share button on the lesson preview, but only once copied (before that its a primary)
4. various links to the Discord

Honestly the first three all feel like they could be primary actions on their own; the one concern with those is that they might clash with the primary button of sign up; I'm not certain for how many / how many percent of users that would actually be a problem.

Less sure to do with the Discord buttons; could just make them `button--clear` as well, but some float a bit in the void and that might look odd:
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/30865f67-cc8b-4a69-8269-10ff64e8b759)

**Comment by KevinMulhern** [comment]
RE: Full curriculum button. A dark color to match the heading text looks like a good option.

<img width="1411" alt="Screenshot 2023-07-10 at 20 14 48" src="https://github.com/TheOdinProject/theodinproject/assets/7963776/164555d4-2447-4a3d-a3be-7d155affb5bb">

**Comment by Asartea** [comment]
> We could put just about any of the other button styles here with how weird the UX is lol. But I'd leave this one for now. I'll be redoing that page when I'm finished with submissions.

You shouldn't give me that much power
![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/4a629c46-e684-4e1e-a1b9-78157ec638b2)

**Comment by KevinMulhern** [comment]
Is it weird I kind like that? 🤣  retrowave vibes.

It's just occurred to me that we're basically getting rid of and replacing the secondary button with the clear button with this. Should we rename the clear button to secondary?

**Comment by Asartea** [comment]
> Is it weird I kind like that? 🤣 retrowave vibes.

I feel like the joke would wear thin rather quickly, although I do want to see the first person encounter this in the wild ;)

> It's just occurred to me that we're basically getting rid of and replacing the secondary button with the clear button with this. Should we rename the clear button to secondary?

Actually yes, that is what we are doing. Sounds good in principle; would have to check if there are any occurrences where we wouldn't want that. 

>RE: Full curriculum button. A dark color to match the heading text looks like a good option.
Could do, but I checked and I can't see a easy drop in replacement in our current set of classes; would probably have to be a mixture of `button--dark` on light mode and `button--gray` on dark mode. Could alternatively also just use `--clear`, which I don't think looks bad, and that would mean we wouldn't need to create an entire new class


![image](https://github.com/TheOdinProject/theodinproject/assets/76259120/6b028a20-d8d0-441d-bae6-a8f574ba04b9)
